src/components/CompareResults/ResultsMain.tsx

```diff
@@ -41,6 +62,33 @@ function ResultsMain() {
     }),
   };
 
+  type SearchParamsType = {
+    [key: string]:
+      | string
+      | string[]
+      | Framework['id']
+      | Framework['name']
+      | Repository['name'];
+  };
+
+  const onFrameworkChange = (event: SelectChangeEvent) => {
+    const id = +event.target.value as Framework['id'];
+    setFrameworkIdVal(id);
+    const searchParamsObject = {} as SearchParamsType;
+    searchParams.forEach((_, key) => {
+      const values = searchParams.getAll(key);
+      // account for revs and repos being arrays
+      if (values.length > 1) {
+        searchParamsObject[key] = values;
+      } else {
+        searchParamsObject[key] = values[0];
+      }
+    });
+
+    setSearchParams({ ...searchParamsObject, framework: id.toString() });
```

Comment:
Can this be replaced with: 
```js
searchParams.set("framework", id.toString());
setSearchParams(searchParams)
```
According to the type in https://reactrouter.com/en/main/hooks/use-search-params, `setSearchParams` accepts a `URLSearchParams` object. I'm passing the same instance here, which doesn't work for react states usually, but I believe this works here because this isn't a state.

Reply:
Oh nice! I mistakenly thought it would change the entire query to only 'framework' rather than preserving the old search params during the update. Good to know!